setup/src/Magento/Setup/Module/Di/Compiler/Log/Writer/Console.php

```diff
@@ -70,6 +70,7 @@ public function write(array $data)
 
         if ($errorsCount) {
             $this->console->writeln('<error>' . 'Total Errors Count: ' . $errorsCount . '</error>');
+            throw new \Magento\Framework\Validator\Exception(__('Error durring compilation'));
```

Comment:
Log writer is incorrect place to throw this exception: nothing went wrong in log writer. Please fix it on command level.